metadata/metadata_test.go

```diff
@@ -207,4 +295,203 @@ var _ = Describe("Metadata", func() {
 			}))
 		})
 	})
+
+	When("CopyAnnotationsWithPrefixReplacement/CopyAnnotationsByPrefix is called", func() {
```

Comment:
Please, split. It's way better to repeat code and keep things cleaner.

Reply:
Updated, thanks!